tensorboard/webapp/widgets/line_chart_v2/sub_view/line_chart_interactive_view.ts

```diff
@@ -0,0 +1,441 @@
+      passive: true,
+    })
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe((event) => {
+        this.dragStartCoord = null;
+        this.isCursorInside = false;
+        this.updateTooltip(event);
+        this.state = InteractionState.NONE;
+        this.changeDetector.markForCheck();
+      });
+
+    fromEvent<MouseEvent>(this.dotsContainer.nativeElement, 'mousemove', {
+      passive: true,
+    })
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe((event) => {
+        switch (this.state) {
+          case InteractionState.SCROLL_ZOOMING: {
+            this.state = InteractionState.NONE;
+            this.updateTooltip(event);
+            this.changeDetector.markForCheck();
+            break;
+          }
+          case InteractionState.NONE:
+            this.updateTooltip(event);
+            this.changeDetector.markForCheck();
+            break;
+          case InteractionState.PANNING: {
+            const deltaX = -event.movementX;
+            const deltaY = -event.movementY;
+            const {width: domWidth, height: domHeight} = this.domDim;
+            const xMin = this.getDataX(deltaX);
+            const xMax = this.getDataX(domWidth + deltaX);
+            const yMin = this.getDataY(domHeight + deltaY);
+            const yMax = this.getDataY(deltaY);
+            this.onViewExtentChange.emit({
+              x: [xMin, xMax],
+              y: [yMin, yMax],
+            });
+            break;
+          }
+          case InteractionState.DRAG_ZOOMING:
+            {
+              if (!this.dragStartCoord) {
+                break;
+              }
+              const xs = [this.dragStartCoord.x, event.offsetX];
+              const ys = [this.dragStartCoord.y, event.offsetY];
+              this.zoomBoxInUiCoordinate = {
+                x: Math.min(...xs),
+                width: Math.max(...xs) - Math.min(...xs),
+                y: Math.min(...ys),
+                height: Math.max(...ys) - Math.min(...ys),
+              };
+            }
+            this.changeDetector.markForCheck();
+            break;
+        }
+      });
+
+    fromEvent<WheelEvent>(this.dotsContainer.nativeElement, 'wheel', {
+      passive: false,
+    })
+      .pipe(
+        takeUntil(this.ngUnsubscribe),
+        switchMap((event: WheelEvent) => {
+          const shouldZoom = !event.ctrlKey && !event.shiftKey && event.altKey;
+          this.showZoomInstruction = !shouldZoom;
+          this.changeDetector.markForCheck();
+
+          if (shouldZoom) {
+            return of(event);
+          }
+          return timer(3000).pipe(
+            tap(() => {
+              this.showZoomInstruction = false;
+              this.changeDetector.markForCheck();
+            }),
+            map(() => null)
+          );
+        }),
+        filter((eventOrNull) => Boolean(eventOrNull))
+      )
+      .subscribe((eventOrNull) => {
+        const event = eventOrNull!;
+        event.preventDefault();
```

Comment:
Wheel should be only prevented default when we should zoom and it was behaving correctly. I understand how it can be very confusing so I moved the preventDefault closer to `shouldZoom` check.